templates/jenkins.service.j2

```diff
@@ -7,10 +7,8 @@ Environment="CASC_JENKINS_CONFIG={{ jenkins_config_dir }}/jenkins.yaml"
 Environment="JENKINS_HOME={{ jenkins_home }}"
 ExecStart={{ jenkins_command }}
 # So systemd won't generate a PID file for non-forking services, so we need to make one.
-# Here we grep only for the WAR filename, not the whole process name, or else quotes and
-# other symbols in the command will cause pgrep to fail. Also, systemd doesn't allow
-# redirection, so we need to wrap the command with sh.
-ExecStartPost=sh -c 'pgrep -f {{ jenkins_war }} > {{ jenkins_pid_file }}'
+# Also, systemd doesn't allow redirection, so we need to wrap the command with sh.
+ExecStartPost=sh -c 'systemctl show jenkins -p ExecMainPID | cut -d = -f 2 > {{ jenkins_pid_file }}'
```

Comment:
In c1071432a62fc21ff9ff250376e84309fd881655 's commits message: "pgrep would returning two" -> "pgrep was returning two"

Reply:
I think I meant "would return two". See bfb55df.